File: src/database/seeders/UserSeeder.php
<?php

namespace Database\Seeders;

use App\Models\User;
use App\Models\Role;
use Illuminate\Database\Seeder;
use Illuminate\Support\Facades\Hash;

class UserSeeder extends Seeder
{
    /**
     * Run the database seeds.
     */
    public function run(): void
    {
        // Obtener los roles
        $adminRole = Role::where('name', 'Admin')->first();
        $empleadoRole = Role::where('name', 'Empleado')->first();
        $consultorRole = Role::where('name', 'Consultor')->first();

        // Crear usuarios dummy
        $users = [
            [
                'name' => 'Administrador',
                'email' => 'admin@ikhana.com',
                'password' => Hash::make('admin123'),
                'role_id' => $adminRole->id,
            ],
            [
                'name' => 'Juan Empleado',
                'email' => 'empleado@ikhana.com',
                'password' => Hash::make('empleado123'),
                'role_id' => $empleadoRole->id,
            ],
            [
                'name' => 'María Consultora',
                'email' => 'consultor@ikhana.com',
                'password' => Hash::make('consultor123'),
                'role_id' => $consultorRole->id,
            ],
        ];

        foreach ($users as $user) {
            User::create($user);
        }
    }
}

File: backend-dotnet/src/Ikhana.Infrastructure/Data/Seeders/UserSeeder.cs
using Ikhana.Application.Common.Interfaces;
using Ikhana.Domain.Entities;
using Microsoft.AspNetCore.Identity;

namespace Ikhana.Infrastructure.Data.Seeders;

public class UserSeeder : IDataSeeder
{
    private readonly UserManager<ApplicationUser> _userManager;

    public int Order => 2;

    public UserSeeder(UserManager<ApplicationUser> userManager)
    {
        _userManager = userManager;
    }

    public async Task SeedAsync(CancellationToken cancellationToken = default)
    {
        var users = new (string Name, string Email, string Password, string Role)[]
        {
            ("Administrador", "admin@ikhana.com", "admin123", "Admin"),
            ("Juan Empleado", "empleado@ikhana.com", "empleado123", "Operador"),
            ("Maria Consultora", "consultor@ikhana.com", "consultor123", "Consultor"),
        };

        foreach (var (name, email, password, role) in users)
        {
            if (await _userManager.FindByEmailAsync(email) is not null)
                continue;

            var user = new ApplicationUser
            {
                UserName = email,
                Email = email,
                Name = name,
                EmailConfirmed = true,
            };

            var result = await _userManager.CreateAsync(user, password);

            if (!result.Succeeded)
                throw new InvalidOperationException(
                    $"Failed to create user {email}: {string.Join(", ", result.Errors.Select(e => e.Description))}");

            await _userManager.AddToRoleAsync(user, role);
        }
    }
}
